src/net.h

```diff
@@ -704,24 +697,44 @@ class CNode
 
     AddrRelay m_addr_relay;
 
-    // inventory based relay
-    CRollingBloomFilter filterInventoryKnown GUARDED_BY(cs_inventory);
-    // Set of transaction ids we still have to announce.
-    // They are sorted by the mempool before relay, so the order is not important.
-    std::set<uint256> setInventoryTxToSend;
     // List of block ids we still have announce.
     // There is no final sorting before sending, as they are always sent immediately
     // and in the order requested.
     std::vector<uint256> vInventoryBlockToSend GUARDED_BY(cs_inventory);
     CCriticalSection cs_inventory;
-    int64_t nNextInvSend{0};
+
+    struct TxRelay {
+        mutable CCriticalSection cs_filter;
+        // We use fRelayTxes for two purposes -
+        // a) it allows us to not relay tx invs before receiving the peer's version message
+        // b) the peer may tell us in its version message that we should not relay tx invs
+        //    unless it loads a bloom filter.
+        bool fRelayTxes GUARDED_BY(cs_filter){false};
+        std::unique_ptr<CBloomFilter> pfilter PT_GUARDED_BY(cs_filter);
+
+        mutable CCriticalSection cs_tx_inventory;
+        CRollingBloomFilter filterInventoryKnown GUARDED_BY(cs_tx_inventory);
```

Comment:
https://github.com/bitcoin/bitcoin/pull/15759/commits/5e6e12ef04b640d32d0341a3a1874fde46b0f95e

Initialize here
```cpp
CRollingBloomFilter filterInventoryKnown GUARDED_BY(cs_tx_inventory){50000, 0.000001};
```
And can drop `.reset()` call, `CRollingBloomFilter` constructor already calls it.

Reply:
> And can drop `.reset()` call, `CRollingBloomFilter` constructor already calls it.

Permalink for convenience: https://github.com/bitcoin/bitcoin/blob/fdc951ad040a9117bc79f12086ce874b4c2aa55a/src/bloom.cpp#L230

Reply:
Thanks, done.